src/templates/template-vision-segmentation/main.py

```diff
@@ -30,8 +39,17 @@ def run(local_rank: int, config: Any):
     # create output folder and copy config file to output dir
     config.output_dir = setup_output_dir(config, rank)
     if rank == 0:
+        #::: if ((it.argparser == 'fire')) { :::#
+        with open(f"{config.output_dir}/config-lock.yaml", "a+") as f:
+            for key, value in config.items():
+                if value is not None:
+                    f.write(f"{key}: {value}\n")
+
+        #::: } else { :::#
         copy(config.config, f"{config.output_dir}/config-lock.yaml")
 
+        #::: } :::#
+
```

Comment:
If now creating a new config-lock.yaml gives the same output as copying with shutil, let's unify two branches into one. Before doing that, please put config-locl.yaml here and original config.yaml for visual comparision.

Reply:
@vfdev-5 It won't give the same output as if we override an argument using cmd it will be stored in new confit file

Reply:
It's ok with overriding the config. My point is that we should be able to restart the training using `config-lock.yaml` same as it is for `argparse` option. It would be also good to reduce the code if possible

Reply:
For `python main.py configs/config.yaml --lr=0.01 --optim=Adam` I got 
```
seed: 666
data_path: ./
train_batch_size: 32
eval_batch_size: 32
num_workers: 4
max_epochs: 20
use_amp: False
debug: False
output_dir: logs/20230815-122747-backend-None-lr-0.0001/20230815-174032-backend-None-lr-0.01
log_every_iters: 2
lr: 0.01
model: resnet18
optim: Adam
config: config.yaml
```
And by using this file we get this config file 
```
seed: 666
data_path: ./
train_batch_size: 32
eval_batch_size: 32
num_workers: 4
max_epochs: 20
use_amp: False
debug: False
output_dir: logs/20230815-122747-backend-None-lr-0.0001/20230815-174032-backend-None-lr-0.01/20230815-174128-backend-None-lr-0.01
log_every_iters: 2
lr: 0.01
model: resnet18
optim: Adam
config: logs/20230815-122747-backend-None-lr-0.0001/20230815-174032-backend-None-lr-0.01/config-lock.yaml

```